src/Modal.js

```diff
@@ -277,6 +277,11 @@ const Modal = React.createClass({
     }
   },
 
+  componentWillUnmount() {
+    this.onHide();
+    this.props.onHide();
```

Comment:
because the modal isn't being hidden it is being unmounted. Which brings up another issue actually. you need to check that `onHide` actually needs to be called. If `props.show` is `false` then it should not be called again.

not calling the onHide prop callback is consistent with every the other components, and with the react ecosystem in general (what I am familiar with of course)